src/treetensornetworks/opsum_to_ttn.jl

```diff
@@ -40,92 +44,79 @@ Hamiltonian, compressing shared interaction channels.
 """
 function ttn_svd(os::OpSum, sites::IndsNetwork, root_vertex; kwargs...)
   # Function barrier to improve type stability
-  coefficient_type = ITensorMPS.determineValType(terms(os))
+  coefficient_type = determine_val_type(terms(os))
   return ttn_svd(coefficient_type, os, sites, root_vertex; kwargs...)
 end
 
-function ttn_svd(
-  coefficient_type::Type{<:Number},
-  os::OpSum,
-  sites0::IndsNetwork,
-  root_vertex;
-  mindim::Int=1,
-  maxdim::Int=typemax(Int),
-  cutoff=eps(real(coefficient_type)) * 10,
-)
-  linkdir_ref = ITensors.In   # safe to always use autofermion default here
+function remove_dups!(v::Vector)
```

Comment:
Is this the same as https://docs.julialang.org/en/v1/base/collections/#Base.unique!?

Reply:
I think it is... will replace and see.

Reply:
I found something interesting here:
* when I used `sort!(v); unique!(v)` instead of `remove_dups!(v)` the tests failed even though `remove_dups!(v)` is doing exactly what the documentation for `unique!(v)` says (plus an initial sort)
* sure enough, I am finding that there are some bugs in `Base.unique!`
* one bug (in my opinion) is that `Base.unique!` calls `Base.isless` on objects in the vector even though the documentation says that `Base.unique!` relies on `isequal` for comparisons. When I input types that do not have `Base.isless` defines, `unique!` errors and does not even run
* the more serious bug is that `unique!` just gives clearly wrong results, and I tracked it down to the `Prod` type from `LazyApply`. I can't quite work out what the issue is yet, and it might be that `Prod` (or `Applied`) has a subtle bug in one of its comparison operators such as `Base.isequal` or `Base.isless` but I haven't found it yet.

So let's keep `remove_dups!` in this PR and treat removing it as a Issue so I can investigate more of what the actual bug is with `unique!` or whether it's a bug with `Prod` etc.

Reply:
Interesting, too bad it doesn't "just work".

Could you rename `remove_dups!` to `_unique!` and put a comment about how we plan to use `Base.unique!` instead but it doesn't work right now for collections of `Prod`?